src/actions/measurements.js

```diff
@@ -0,0 +1,241 @@
+      newControls.measurementsGroupBy = group_by;
+    }
+    if (measurement_display) {
+      newControls.measurementsDisplay = measurement_display;
+    }
+    if (typeof show_overall_mean === "boolean") {
+      newControls.measurementsShowOverallMean = show_overall_mean;
+    }
+    if (typeof show_threshold === "boolean") {
+      newControls.measurementsShowThreshold = show_threshold;
+    }
+  }
+
+  return newControls;
+};
+
+export const loadMeasurements = (json) => (dispatch, getState) => {
+  // TODO: Load controls from state to get potential url query parameters
+  const { tree, controls } = getState();
+  if (!tree.loaded) {
+    throw new Error("tree not loaded");
+  }
+
+  const collections = json["collections"];
+  if (!collections || collections.length === 0) {
+    throw new Error("Measurements JSON does not have collections");
+  }
+
+  collections.forEach((collection) => {
+    /*
+     * Create fields Map for easier access of titles and to keep ordering
+     * First add fields from JSON to keep user's ordering
+     * Then loop over measurements to add any remaining fields
+     */
+    const collectionFieldsArray = collection.fields;
+    collection.fields = new Map();
+    if (collectionFieldsArray && collectionFieldsArray.length > 0) {
+      collectionFieldsArray.forEach(({key, title}) => {
+        collection.fields.set(key, {title: title || key});
+      });
+    }
+
+    /**
+     * Create filters Map for easier access of values and to keep ordering
+     * First add fields from JSON to keep user's ordering
+     * Then loop over measurements to add values
+     * If there are no JSON defined filters, then add all fields as filters
+     */
+    const collectionFiltersArray = collection.filters;
+    collection.filters = new Map();
+    if (collectionFiltersArray && collectionFiltersArray.length > 0) {
+      collectionFiltersArray.forEach((filterField) => {
+        collection.filters.set(filterField, {values: new Set()});
+      });
+    }
+
+    collection.measurements.forEach((measurement, index) => {
+      Object.entries(measurement).forEach(([field, fieldValue]) => {
+        // Add remaining field titles
+        if (!collection.fields.has(field)) {
+          collection.fields.set(field, {title: field});
+        }
+
+        // Only save the unique values if the field is in defined filters
+        // OR there are no JSON defined filters, so all fields are filters
+        if ((collection.filters.has(field)) || !collectionFiltersArray) {
+          const filterObject = collection.filters.get(field) || { values: new Set()};
+          filterObject.values.add(fieldValue);
+          collection.filters.set(field, filterObject);
+        }
+      });
+
+      // Add jitter and stable id for each measurement to help visualization
+      // Generate Gaussian jitter with a Box-Muller transform
+      measurement["measurementJitter"] = Math.sqrt(-2*Math.log(Math.random()))*Math.cos(2*Math.PI*Math.random());
```

Comment:
@joverlee521 [here's the jitter code currently in use by the scatterplot layout](https://github.com/nextstrain/auspice/blob/e48e4326240740bc9a1053f1f71849e89e597d85/src/components/tree/phyloTree/layouts.js#L507-L527), which could potentially be refactored for the measurements. It would be nice for both approaches to use the same algorithm at least. 

I precomputed random values as this turned out to be a speed bottleneck.

Reply:
Addressed in bb2b19a, I went with a slightly simpler jitter generation.